File: src/common/py/data/entities/project/history/publishing_history_record.py
from dataclasses import dataclass
from enum import IntEnum

from dataclasses_json import dataclass_json

from .project_history_record import ProjectHistoryRecord
from ...connector import ConnectorInstanceID


@dataclass_json
@dataclass
class PublishingHistoryRecord(ProjectHistoryRecord):
    """
    A single record of a project's publishing history.

    Attributes:
        status: The status (done or failed).
        message: An optional message (usually in case of an error).
    """

    class Status(IntEnum):
        """
        The status of the publishing operation.
        """

        DONE = 0
        FAILED = -1

    connector_instance: ConnectorInstanceID

    status: Status
    message: str

File: src/common/web/data/entities/project/history/PublishingHistoryRecord.ts
/*
 * The status of the publishing operation.
 */
import { type ConnectorInstanceID } from "../../connector/ConnectorInstance";
import { ProjectHistoryRecord } from "./ProjectHistoryRecord";

export const enum PublishingHistoryRecordStatus {
    Done = 0,
    Failed = -1,
}

/**
 * A single record of a project's publishing history.
 *
 * @param status - The status (done or failed).
 * @param message - An optional message (usually in case of an error).
 */
export class PublishingHistoryRecord extends ProjectHistoryRecord {
    public readonly connector_instance: ConnectorInstanceID;

    public readonly status: PublishingHistoryRecordStatus;
    public readonly message: string;

    public constructor(timestamp: number, connectorInstance: ConnectorInstanceID, status: PublishingHistoryRecordStatus, message: string) {
        super(timestamp);

        this.connector_instance = connectorInstance;

        this.status = status;
        this.message = message;
    }
}
